File: test/wavetables/c_order_multi_dim.py
import numpy as np
N=10
x=np.zeros((N,N,N),dtype='float32')
for i in range(N):
    for j in range(N):
        for k in range(N):
            x[i,j,k]=N*N*i+N*j+k
x.tofile('/tmp/multi_dims.f32')

File: test/wavetables/c_order_multi_dim.c
#include <stdio.h>
#define N 10

int main(void)
{
    float x[N*N*N];
    FILE *f = fopen("/tmp/multi_dims.f32","r");
    fread(x,sizeof(float),N*N*N,f);
    int i,j,k;
    for (i = 0; i < N; i++) {
        for (j = 0; j < N; j++) {
            for (k = 0; k < N; k++) {
                printf("%f\n",x[N*N*i+N*j+k]);
            }
        }
    }
    return 0;
}
